staff/ismael-haytam/pre-course/tema-1/Methods.js

```diff
@@ -0,0 +1,335 @@
+
+function deleteElementFromArray(infos) {
+    infos.forEach((info, i) => (info === 'Girona') ? infos.splice(i, 1) : '');
+    return console.log(`City deleted! => ${infos.join(', ')}`);
+}
+
+deleteElementFromArray(['Ismael Haytam', 'Tanane', 22, 'Girona']);
+
+
+
+
+
+// j) Ahora, elimina el nombre y asegura los cambios
+
+function deleteNameFromArray(infos) {
+    infos.forEach((info, i) => (info === 'Ismael Haytam') ? infos.splice(i, 1) : '');
+    return console.log(`Name deleted! => ${infos.join(', ')}`);
+}
+
+deleteNameFromArray(['Ismael Haytam', 'Tanane', 22, 'Girona']);
+
+
+
+
+
+// k) Quiero volver a introducir mi nombre, como podria hacer para introducirlo en la primera posición?
+
+function unshiftName(infos, name) {
+    infos.unshift(name);
+    return console.log(`Name added in the first position! => ${infos.join(' ')}`);
+}
+
+unshiftName(['Tanane', 22, 'Girona'], 'Ismael Haytam');
+
+
+
+
+
+// l) Ahora, declara una array con los números del 0 a 10 y muestra cada número multiplicado por dos.
+
+let multiplyNumber = numbers => console.log(numbers.map(number => (number * 2)).join(' '));
+
+multiplyNumber([0,1,2,3,4,5,6,7,8,9,10]);
+
+
+
+
+
+// var num = 3; // cada número se multiplicará por 3
+
+let multiplyWithSpecificNumber = (numbers, mul) => console.log(numbers.map(number => (number * mul)).join(' '));
+
+multiplyWithSpecificNumber([0,1,2,3,4,5,6,7,8,9,10], 3);
+
+
+
+
+
+// m) Podrías mostrarlos en el orden inverso?
+
+let reverseArray = numbers => console.log(numbers.reverse().join(' '));
+
+reverseArray([0,1,2,3,4,5,6,7,8,9,10]);
+
+
+
+
+
+// n1) Puedes indicarme que letras se repiten de tu nombre y cuantas veces?
+
+function countRepetitions(name) {
+    let result = [];
+    name.toUpperCase().split(' ').map(word => {
+        word.split('').forEach(letter => {
+            result.push(`'${letter}' => ${name.match(new RegExp(letter, 'gi')).length} times`);
+            result = [...new Set(result)];
+        });
+    });
+    return console.log(result.join(' '));
+}
+
+countRepetitions('Ismael Haytam Tanane');
+
+
+
+
+
+// n1) Ahora muestra por consola que letras NO se repiten y muestra tu nombre sin esas letras
+
+function showUnreperited(name) {
+    let result = [];
+    name.toUpperCase().split(' ').map(word => word.split('').forEach(letter => {
+        name.match(new RegExp(letter, 'gi')).length === 1 ? result.push(letter) : '';
+    }));
+    return console.log(`The letters => ${result.join(', ')}  - the name is => ${result.join('')}`);
+}
+
+showUnreperited('Ismael Haytam Tanane');
+
+
+
+
+/*
+    -   Numbers ---------------------------------------------------------
+*/
+
+// a) Que hora es? Declara la hora como número y devuelvela como String
+
+let currentTime = (time) => console.log(`I'ts ${time.toString()} of morning`);
+
+currentTime(10.45);
+
+
+
+
+
+// b) Nono, que hora exactamente? Dime la hora sin minutos!
+
+let justHour = (time) => console.log(`I'ts ${parseInt(time)} of morning`);
```

Comment:
are parenthesis required in (time)?

Reply:
not required if there is only one argument